File: frontend/src/types/teamPost.ts
export interface TeamPost {
  id: number;
  team_id: number;
  author_id: number;
  author_name: string;
  author_role: string;
  author_photo_url?: string | null;
  content: string;
  media_url?: string | null;
  created_at: string;
}

File: backend/app/schemas/team_post.py
from datetime import datetime

from pydantic import ConfigDict
from sqlmodel import SQLModel


class TeamPostBase(SQLModel):
    content: str
    media_url: str | None = None


class TeamPostCreate(SQLModel):
    content: str


class TeamPostRead(TeamPostBase):
    id: int
    team_id: int
    author_id: int
    author_name: str
    author_role: str
    author_photo_url: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
